test/Parse/switch.swift

```diff
@@ -276,6 +276,24 @@ func patternVarDiffType(x: Int, y: Double) {
   }
 }
 
+func patternVarDiffMutability(x: Int, y: Double) {
+  switch x {
+  case let a where a < 5, var a where a > 10: // expected-error {{'var' pattern binding must match previous 'let' pattern binding}}{{27-30=let}}
+    break
+  default:
+    break
+  }
+  switch (x, y) {
+  // Would be nice to have a fixit in the following line if we detect that all bindings in the same pattern have the same problem.
+  case let (a, b) where a < 5, var (a, b) where a > 10: // expected-error 2{{'var' pattern binding must match previous 'let' pattern binding}}{{none}}
+    break
+  case (let a, var b) where a < 5, (let a, let b) where a > 10: // expected-error {{'let' pattern binding must match previous 'var' pattern binding}}{{44-47=var}}
```

Comment:
Can you add a test case with three patterns?

Reply:
@jrose-apple Ah, good call, with three patterns and two wrong, invalidating the first pattern variable when comparing with the 2nd caused a spurious diagnostic about the type of the 3rd not matching error-type. Fixed now!